xds/src/main/java/io/grpc/xds/XdsLrsClient.java

```diff
@@ -0,0 +1,325 @@
+  private LrsStream lrsStream;
+
+  XdsLrsClient(ManagedChannel channel, Helper helper, Supplier<Stopwatch> stopwatchSupplier,
+      BackoffPolicy.Provider backoffPolicyProvider) {
+    this.channel = checkNotNull(channel, "channel");
+    this.helper = checkNotNull(helper, "helper");
+    this.serviceName = checkNotNull(helper.getAuthority(), "serviceName");
+    this.syncContext = checkNotNull(helper.getSynchronizationContext(), "syncContext");
+    this.stopwatchSupplier = checkNotNull(stopwatchSupplier, "stopwatchSupplier");
+    this.stopwatch = stopwatchSupplier.get();
+    this.logger = checkNotNull(helper.getChannelLogger(), "logger");
+    this.timerService = checkNotNull(helper.getScheduledExecutorService(), "timeService");
+    this.backoffPolicyProvider = checkNotNull(backoffPolicyProvider, "backoffPolicyProvider");
+  }
+
+  // TODO (chengyuanzhang): consider putting to a util class.
+  @VisibleForTesting
+  static long durationToNum(Duration duration, TimeUnit unit) {
+    return unit.convert(duration.getSeconds() * 1000_000_000L + duration.getNanos(),
+        TimeUnit.NANOSECONDS);
+  }
+
+  void startLrsRpc() {
+    checkState(lrsStream == null, "previous lbStream has not been cleared yet");
+    LoadReportingServiceGrpc.LoadReportingServiceStub stub
+        = LoadReportingServiceGrpc.newStub(channel);
+    lrsStream = new LrsStream(stub, stopwatchSupplier.get());
+    lrsStream.start();
+    stopwatch.reset().start();
+    lrsStream.stopwatch.reset().start();
+
+    LoadStatsRequest initRequest =
+        LoadStatsRequest.newBuilder()
+            .setNode(Node.newBuilder()
+                .setMetadata(Struct.newBuilder()
+                    .putFields(
+                        TRAFFICDIRECTOR_GRPC_HOSTNAME,
+                        Value.newBuilder().setStringValue(serviceName).build())))
+            .build();
+    try {
+      lrsStream.lrsRequestWriter.onNext(initRequest);
+    } catch (Exception e) {
+      lrsStream.close(e);
+    }
+  }
+
+  void shutdownLrsRpc() {
+    if (lrsStream != null) {
+      lrsStream.close(null);
+    }
+  }
+
+  void cancelLrsRpcRetryTimer() {
+    if (lrsRpcRetryTimer != null) {
+      lrsRpcRetryTimer.cancel();
+    }
+  }
+
+  void shutdown() {
+    shutdownLrsRpc();
+    cancelLrsRpcRetryTimer();
+    // Do not shutdown channel as it is not owned by LrsClient.
+  }
+
+  @VisibleForTesting
+  static class LoadReportingTask implements Runnable {
+    private final LrsStream stream;
+
+    LoadReportingTask(LrsStream stream) {
+      this.stream = stream;
+    }
+
+    @Override
+    public void run() {
+      stream.loadReportTimer = null;
+      stream.sendLoadReport();
+    }
+  }
+
+  @VisibleForTesting
+  class LrsRpcRetryTask implements Runnable {
+
+    @Override
+    public void run() {
+      startLrsRpc();
+    }
+  }
+
+  // TODO (chengyuanzhang): consider putting to a util class.
+  @VisibleForTesting
+  static Duration numToDuration(long duration, TimeUnit unit) {
```

Comment:
You should use `com.google.protobuf.util.Durations` too.

Reply:
ACK.